com.unity.render-pipelines.universal/Editor/Decal/DecalProjectorEditor.cs

```diff
@@ -0,0 +1,772 @@
+                uvHandles.DrawHandle();
+                if (EditorGUI.EndChangeCheck())
+                {
+                    Undo.RecordObject(decalProjector, "Decal Projector Change");
+
+                    for (int channel = 0; channel < 2; channel++)
+                    {
+                        // Preserve serialized state for axes with the scaled size 0.
+                        if (scaledSize[channel] != 0f)
+                        {
+                            float handleSize = uvHandles.size[channel];
+                            float minusNewUVStart = .5f * handleSize - uvHandles.center[channel];
+                            float decalSize = decalProjector.size[channel];
+                            float limit = k_LimitInv * decalSize;
+                            if (handleSize > limit || handleSize < -limit)
+                            {
+                                uvScale[channel] = decalSize / handleSize;
+                                uvBias[channel] = minusNewUVStart / handleSize;
+                            }
+                            else
+                            {
+                                // TODO: Decide if uvHandles.size should ever have negative value. It can't currently.
+                                uvScale[channel] = k_Limit * Mathf.Sign(handleSize);
+                                uvBias[channel] = k_Limit * minusNewUVStart / decalSize;
+                            }
+                        }
+                    }
+
+                    decalProjector.uvScale = uvScale;
+                    decalProjector.uvBias = uvBias;
+                }
+            }
+        }
+
+        void DrawHandles()
+        {
+            DecalProjector decalProjector = target as DecalProjector;
+
+            if (editMode == k_EditShapePreservingUV || editMode == k_EditShapeWithoutPreservingUV)
+                DrawBoxTransformationHandles(decalProjector);
+            else if (editMode == k_EditUVAndPivot)
+            {
+                DrawPivotHandles(decalProjector);
+                DrawUVHandles(decalProjector);
+            }
+        }
+
+        [DrawGizmo(GizmoType.Selected | GizmoType.Active)]
+        static void DrawGizmosSelected(DecalProjector decalProjector, GizmoType gizmoType)
+        {
+            UpdateColorsInHandlesIfRequired();
+
+            const float k_DotLength = 5f;
+
+            // Draw them with scale applied to size and pivot instead of the matrix to keep the proportions of the arrow and lines.
+            using (new Handles.DrawingScope(fullColor, Matrix4x4.TRS(decalProjector.transform.position, decalProjector.transform.rotation, Vector3.one)))
+            {
+                Vector3 scale = decalProjector.effectiveScale;
+                Vector3 scaledPivot = Vector3.Scale(decalProjector.pivot, scale);
+                Vector3 scaledSize = Vector3.Scale(decalProjector.size, scale);
+
+                boxHandle.center = scaledPivot;
```

Comment:
Not sure but I guess for the decal Editor you have duplicated  a lot of code with HDRP, stuff like gizmo and others should be share. IIRC you mention you will do that in next PR?

Reply:
One worry is that we have done several fixes for HDRP in the decal widget and I am afraid you willl not benefit from it.

Reply:
cc @RSlysz for awareness

Reply:
Also the reverse is true, if during your PR work you improve stuff in this code, HDRP don't benefit from it

Reply:
Yes that is planned for follow up. I think any additional changes can be included in follow up once it moved to core

Reply:
It would have been simpler to move the gizmo to CoreRP IMO.
Please add me on the review while merging stuffs.

Reply:
PR is already huge and we are going to have quite a lot of landing and conflicts. I want to avoid touching core/hdrp in this pr